src/Data/Singletons/Partition.hs

```diff
@@ -58,15 +59,17 @@ partitionDec :: DsMonad m => DDec -> m PartitionedDecs
 partitionDec (DLetDec (DPragmaD {})) = return mempty
 partitionDec (DLetDec letdec) = return $ mempty { pd_let_decs = [letdec] }
 
-partitionDec (DDataD nd _cxt name tvbs cons derivings) = do
+partitionDec (DDataD nd _cxt name tvbs k cons derivings) = do
+  extra_tvbs <- mkExtraDKindBinders k
```

Comment:
This should use `buildDataDType`, no?

Reply:
I pondered this, but we end up needing intermediate parts of the `buildDataDType` computation (e.g., `all_tvbs` in `data_dec` and `ty` in `derived_decs`), so I decided to just inline the definition here to avoid some redundant work.